client/landing/gutenboarding/stores/onboard/reducer.ts

```diff
@@ -200,6 +200,19 @@ const selectedFeatures: Reducer< FeatureId[], OnboardAction > = (
 	return state;
 };
 
+const randomizedDesigns: Reducer< { featured: Design[] }, OnboardAction > = (
+	state = { featured: [] },
+	action
+) => {
+	if ( action.type === 'SET_RANDOMIZED_DESIGNS' ) {
+		return action.randomizedDesigns;
+	}
+	if ( action.type === 'RESET_ONBOARD_STORE' ) {
+		return { featured: [] };
+	}
```

Comment:
I wasn't 100% sure that a reset should set `{ featured: []}`.  Does that seem correct to you?

Reply:
In this case, because the calling code always expects an object with the shape `{ featured: [] }`, setting the reset to `{ featured: [] }` does seems okay to me, but if we were to add other keys at some point, this reducer might get more complex.

Another way to do it if you want to set the default and reset state to `[]` would be to use nested `combineReducer`s. So you might have something like:

```
// Rename the existing randomizedDesigns reducer to featuredRandomizedDesigns and then add:

const randomizedDesigns = combineReducers( { featured: featuredRandomizedDesigns } ); 
```

If you went with this approach, then in your reducer you'd return `action.randomizedDesigns?.featured`;

Then in the `randomizedDesigns` function you can just deal with the array, without having to worry about the `featured` key. There's an example of this kind of reducer nested / combined reducers in https://github.com/automattic/wp-calypso/blob/HEAD/packages/data-stores/src/site/reducer.ts#L91 (e.g. take a look at the `newSite` reducer there).

---

That said, that's just one way to handle it! Setting the default and reset state to `{ featured: [] }` is totally okay to begin with. Personally, I'm quite a fan of iterative changes, so whichever you choose to go with here I think will be fine.

Reply:
I'll go ahead and leave this as is, with reset defaulting to `{ featured: [] }`